### PR TITLE
upgrade elasticsearch/elasticsearch to 2.0

### PR DESCRIPTION
**Pull request by jdohuutin**
Upgrade `elasticsearch/elasticsearch` to 2.0.

(see https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_breaking_changes_from_1_x.html).

- use `ClientBuilder::fromConfig` for elasticsearch client instanciation.
- remove `CurlMultiConnectionDecorator` and `GuzzleConnectionDecorator`
- fire ElasticsearchEvent with `EventHandler`
- add custom request headers with `HeadersHandler` (like `Accept-Encoding`)

Somes features might miss like connection pool configuration, custom logging, old custom guzzle options..

### CODE REVIEW

File: src/M6Web/Bundle/ElasticsearchBundle/DependencyInjection/M6WebElasticsearchExtension.php [hunk 1]
```diff
@@ -22,86 +22,100 @@ class M6WebElasticsearchExtension extends Extension
...
      * @return string
      */
-    public function getAlias()
+    protected function createHandler(ContainerBuilder $container, array $config, $definitionId)
     {
-        return 'm6web_elasticsearch';
-    }
+        // cURL handler
+        $singleHandler   = (new Definition('GuzzleHttp\Ring\Client\CurlHandler'))
+            ->setPublic(false)
```
Comment (omansour) [inline src/M6Web/Bundle/ElasticsearchBundle/DependencyInjection/M6WebElasticsearchExtension.php]: :+1:

**Comment by mguillermin** [comment]
The data collector is not working anymore (I'm always seeing 0 queries). 

I'm working on a fix using an event listener instead of a Logger. I will add it to this PR when ready.

**Comment by mguillermin** [comment]
I've updated the data collector. It seems to work well now and I've added a few informations in the collector view.
I also added a configuration option to be able to set a custom Logger.

**Comment by jdohuutin** [comment]
:+1: good work with the data collector.

**Comment by omansour** [comment]
:+1:

**Comment by mguillermin** [comment]
When we will merge this PR, should we just tag this 1.0 by simply following semver or is it better to use the https://github.com/elastic/elasticsearch-php major version (2.x) ?

What do you think ?

**Comment by t-geindre** [comment]
:+1:

File: src/M6Web/Bundle/ElasticsearchBundle/DependencyInjection/M6WebElasticsearchExtension.php [hunk 2]
```diff
@@ -22,86 +22,135 @@ class M6WebElasticsearchExtension extends Extension
     public function load(array $configs, ContainerBuilder $container)
     {
         $configuration = new Configuration();
-        $config = $this->processConfiguration($configuration, $configs);
+        $config        = $this->processConfiguration($configuration, $configs);
```
Comment (mikaelrandy) [inline src/M6Web/Bundle/ElasticsearchBundle/DependencyInjection/M6WebElasticsearchExtension.php]: I think we should not align assigment 'cause if you add a new variable with name lenth higher than existing ones, you have to reindent whole block and made some noise on diff

**Comment by mikaelrandy** [comment]
Seems good, bu we have to create a "v1" branch on master before merging this PR to keep availability to fix ols code.

**Comment by jdohuutin** [comment]
branch 1.0 https://github.com/M6Web/ElasticsearchBundle/tree/1.0

**Comment by jdohuutin** [comment]
Release v1.0.0 https://github.com/M6Web/ElasticsearchBundle/releases/tag/v1.0.0

**Pull request merged**